### DISCUSSION TITLE
Ability to chat within the same network, avoid sharing long room name IDs and passwords

### DISCUSSION THREAD

**Discussion by OutshineIssue** [jeremyckahn/chitchatter#244] [Ideas]
You know how you can just open [snapdrop.net](https://www.snapdrop.net) on two devices on the same network and start sending files immediately? Imagine no room name, no password, frictionless chat. Of course, it's not everyone's use case to chat over the same network, but it's a nice feature to have, and it's probably easy to implement since [Snapdrop](https://github.com/RobinLinus/snapdrop) uses almost the same technologies as ChitChatter.

**Comment by jeremyckahn** [comment]
Hi @OutshineIssue, thanks for sharing this idea. It's not clear how this might work, at least with [Trystero](https://github.com/dmotz/trystero) (the library the handles networking for Chitchatter). If Trystero adds support for local networking in the future, I'd be open to considering how Chitchatter might incorporate it.

**Reply by OutshineIssue** [reply]
Seems fair. This may be for another discussion, but what about room names that can be easily shared "verbally"? Maybe you've heard of File.pizza, their shared links look like this

> https://file.pizza/romano/shallots/tomatoes/scallions

Although it's far from perfect...

As a non-native English speaker, I have no idea what a shallots are, and romano pizzas are not really common where I live, so I would have a hard time writing those, but it's better than 'd67fna48-something-something'. If you decide to give this idea a try, maybe use a pool of words that could be easily identifiable for non-natives English speakers, like the top 500 most common words in the English dictionary, PLUS the less characters per word, the better.

**Comment by jeremyckahn** [comment]
> but what about room names that can be easily shared "verbally"?

I think that would be a good addition to Chitchatter! It seems like something that would be a high-level, "pure UI" change since room names are arbitrary. I don't have plans to work on this, but I'd be open to accepting a pull request that implements it.
